File: Classes/EVEOnlineAPI/Character/EVEContactList.m
//
//  EVEContactList.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 6/2/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEContactList.h"

@implementation EVEContactListContact

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"contactID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"contactName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"standing":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"contactTypeID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"labelMask":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"inWatchlist":@{@"type":@(EVEXMLSchemePropertyTypeScalar)}};
	return scheme;
}

@end

@implementation EVEContactListLabel

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"labelID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"name":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	return scheme;
}

@end

@implementation EVEContactList

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"contactList":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEContactListContact class]},
				   @"contactLabels":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEContactListLabel class]},
				   @"corporateContactList":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEContactListContact class]},
				   @"corporateContactLabels":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEContactListLabel class]},
				   @"allianceContactList":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEContactListContact class]},
				   @"allianceContactLabels":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEContactListLabel class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEContactList.swift
//
//  EVEContactList.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 29.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEContactListItem: EVEObject {
	public var contactID: Int = 0
	public var contactName: String = ""
	public var standing: Int = 0
	public var contactTypeID: Int = 0
	public var labelMask: Int = 0
	public var inWatchlist: Bool = false
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"contactID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"contactName":EVESchemeElementType.String(elementName:nil, transformer:nil),
			"standing":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"contactTypeID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"labelMask":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"inWatchlist":EVESchemeElementType.Bool(elementName:nil, transformer:nil),
		]
	}
}

public class EVEContactListLabel: EVEObject {
	public var labelID: Int = 0
	public var name: String = ""
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"labelID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"name":EVESchemeElementType.String(elementName:nil, transformer:nil),
		]
	}
}

public class EVEContactList: EVEResult {
	public var contactList: [EVEContactListItem] = []
	public var contactLabels: [EVEContactListLabel] = []
	public var corporateContactList: [EVEContactListItem] = []
	public var corporateContactLabels: [EVEContactListLabel] = []
	public var allianceContactList: [EVEContactListItem] = []
	public var allianceContactLabels: [EVEContactListLabel] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"contactList":EVESchemeElementType.Rowset(elementName: nil, type: EVEContactListItem.self, transformer: nil),
			"contactLabels":EVESchemeElementType.Rowset(elementName: nil, type: EVEContactListLabel.self, transformer: nil),
			"corporateContactList":EVESchemeElementType.Rowset(elementName: nil, type: EVEContactListItem.self, transformer: nil),
			"corporateContactLabels":EVESchemeElementType.Rowset(elementName: nil, type: EVEContactListLabel.self, transformer: nil),
			"allianceContactList":EVESchemeElementType.Rowset(elementName: nil, type: EVEContactListItem.self, transformer: nil),
			"allianceContactLabels":EVESchemeElementType.Rowset(elementName: nil, type: EVEContactListLabel.self, transformer: nil),
		]
	}
}
